include/elm/inhstruct/DLList.h

```diff
@@ -31,31 +31,38 @@ class DLNode {
 	friend class DLList;
 	DLNode *nxt, *prv;
 public:
+	DLNode() : nxt(nullptr), prv(nullptr) {}
+
 	inline DLNode *next(void) const { return nxt; }
 	inline DLNode *previous(void) const { return prv; }
-	inline bool atBegin(void) const { return prv == 0; }
-	inline bool atEnd(void) const { return nxt == 0; }
+	inline bool atBegin(void) const { return prv == nullptr; }
+	inline bool atEnd(void) const { return nxt == nullptr; }
 	
 	inline void replace(DLNode *node) {
 		ASSERTP(node, "null node for replacement");
-		nxt->prv = node; node->nxt = nxt;
-		prv->nxt = node; node->prv = prv;
+		if (nxt) nxt->prv = node;
```

Comment:
If I understand correctly, the DLNode data structure relies on `nxt` and `prev` always existing at these points. This condition would only be triggered when the struct is corrupted, and it works around a crash, but it further corrupts the struct (for example, by running only part of the necessary `DLNode::replace(...)` instructions...

I am not convinced this does anything but hide the errors

Reply:
Here as well, I'm simply trying to ensure that each pointer is properly initialized and not nullptr before attempting to use a method or attribute.